frontend/src/component/EmployeeForm.tsx

```diff
@@ -0,0 +1,214 @@
+            added,
+        }
+        if (props.employee.id) {
+            editEmployee.id = props.employee.id
+        }
+
+        if (file) {
+            props.onSubmit(editEmployee, file)
+            if (props.navigateTo) {
+                navigate(props.navigateTo)
+            }
+        } else {
+            props.onSubmit(editEmployee)
+            if (props.navigateTo) {
+                navigate(props.navigateTo)
+            }
+        }
+    }
+
+    function handleCancelButton() {
+        if (props.navigateTo) {
+            navigate(props.navigateTo)
+        }
+    }
+
+    return (
+        <>
+            <form onSubmit={formSubmitHandler} className={"add-employee"}>
+                <div>
+                    <label htmlFor="firstname">First Name:</label>
+
+                    <input type={"text"} id="firstName" name="firstname" onChange={handleFirstNameChange}
+                           value={firstName} placeholder={"firstname"}
+                           required={true}/>
+                </div>
+                <div>
+                    <label htmlFor={lastName}> Last Name: </label>
+                    <input type={"text"} id="lastname" name="lastname" onChange={handleLastNameChange} value={lastName}
+                           placeholder={"lastname"}
+                           required={true}/>
+                </div>
+                <div>
+                    <label>Position:</label>
+                    <input type={"text"} onChange={handlePositionChange} value={position} placeholder={"position"}
+                           required={true}/>
+                </div>
+                <div>
+                    <label>Date of birth: </label>
+                    <input type={"date"} onChange={handleDateOfBirthChange} value={dateOfBirth}
+                           placeholder={"date of birth"} required={true}/>
+                </div>
+                <div>
+                    <label>
+                        Address:
+                        <input type={"text"} onChange={handleAddress_StreetChange} value={address.street}
+                               placeholder={"street"} required={true}/>
+                        <input type={"text"} onChange={handleAddress_HouseNumberChange} value={address.houseNumber}
+                               placeholder={"house number"} required={true}/>
+                        <input type={"text"} onChange={handleAddress_PostalCodeChange} value={address.postalCode}
+                               placeholder={"postal code"} required={true}/>
+                        <input type={"text"} onChange={handleAddress_CityChange} value={address.city}
+                               placeholder={"city"}
+                               required={true}/>
+                    </label>
+                </div>
+                <label>
+                    E-Mail:
+                    <input type={"text"} onChange={handleEmailChange} value={email} placeholder={"e-mail"}
```

Comment:
You could check out the different possible input types. There is for example one for email  :)